Gateway/Data/Order/OrderAdapter.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Braintree\Gateway\Data\Order;
+
+use Magento\Payment\Gateway\Data\AddressAdapterInterface;
+use Magento\Payment\Gateway\Data\Order\AddressAdapter;
+use Magento\Payment\Gateway\Data\Order\AddressAdapterFactory;
+use Magento\Payment\Gateway\Data\OrderAdapterInterface;
+use Magento\Sales\Model\Order;
+
+/**
+ * Class OrderAdapter
+ */
+class OrderAdapter implements OrderAdapterInterface
+{
+    /**
```

Comment:
You could extend the original OrderAdapter which'd implement the core requried methods instad of redeclaring